jax/experimental/jax2tf/jax2tf.py

```diff
@@ -103,67 +103,79 @@ def add_unit(v: TfValOrUnit, aval: core.AbstractValue):
 tf_impl: Dict[core.Primitive,
               Callable[..., Any]] = {}
 
-def disable_gradient(fun):
-  """Prevents the wrapped function from being differentiated."""
-
-  def grad_disabled(*dy, variables=None):
-    raise ValueError("jax2tf currently does not support gradients. Please "
-                     "reach out to jax-core@ if this feature is a blocker "
-                     "for you, we are working on it!")
-
-  is_tensor = lambda t: isinstance(t, (tf.Tensor, tf.Variable))
-
-  def wrapper(*args, **kwargs):
-    flat_args, treedef = jax.tree_flatten((args, kwargs))
-    tensor_idxs = [i for i, t in enumerate(flat_args) if is_tensor(t)]
-    tensor_args = [t for t in flat_args if is_tensor(t)]
-
-    @tf.custom_gradient
-    def inner_wrapper(*tensor_args):
-      flat_copy = list(flat_args)
-      for i, t in zip(tensor_idxs, tensor_args):
-        flat_copy[i] = t
-
-      args, kwargs = jax.tree_unflatten(treedef, flat_copy)
-      out = fun(*args, **kwargs)
-      return out, grad_disabled
-
-    return inner_wrapper(*tensor_args)
-
-  return wrapper
-
-
-def convert(fun):
+def convert(fun,
+            with_gradient=False):
```

Comment:
I feel like `True` might be a safer default to start with? Also nit that we don't need to split the line here.

Reply:
Will fix the newline. 

I did think about the default, and I see your point. I thought of disabling gradients by default because it leads to somewhat better behaviour once you use a savedmodel. Currently a restored saved model that contains custom gradients fails in obscure ways. If we disable the gradients, we add a PreventGradient op, which is saved in saved model and gives a nice error message. 

Once we fix the handling of gradients in savedmodel, even if by inserting a PreventGradient in lieu of a custom gradient, then I think we can change the default. That is a safe change, because people who did not pass the `with_gradient` will then find themselves with code that supports AD but their code certainly does not rely on gradients or else they would have had errors.